fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -90,4 +90,22 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
+	 * (This is also known as "recipe remainder".)
+	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
+	 *
+	 * <p>The passed {@code stack} should not be modified, and its count should be ignored.
+	 * The returned stack, if not empty, must be a new instance with count 1.
+	 *
+	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
+	 * Using {@link FabricItemStack#hasRecipeRemainder} and {@link FabricItemStack#getRecipeRemainder} instead of calling this function directly is recommended.
```

Comment:
Sorry if I seem pedantic, i think this must be better and more explicit (about the potentially issues)
And... the mentioned functions does not exists, or remaind to the same method that is unclear (i mean the `FabricItemStack.getRecipeRemainder()`.

```suggestion
	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
	 *
	 * <p>Reimplementing this method must follow this "best and only practice rule"
	 * to avoid compatibility issues with other mods that use recipe remainders:
	 * You must reimplement this only if you are sure that the stack has a maximum count of 1,
	 * and the returned stack must have the same count of the input stack.
	 * Only exception is the empty stack that has a count of zero.
	 *
	 * <p>For other use cases, is highly recommended to set the remainder with {@link Item.Settings#recipeRemainder(Item)}
	 * instead of reimplementing this.
```